services/meta/ota.md

```diff
@@ -7,34 +7,44 @@ This service is used to manage over-the-air (OTA) upgrades of end devices.
 
 #### Interfaces
 
-Type | Interface               | Value type | Description
------|-------------------------|------------|--------------
-out  | evt.ota_progress.report | int_map    | Shows upgrade progress and estimated time left.
-out  | evt.ota_end.report      | object     | Sent on upgrade end with upgrade status.
+Type | Interface               | Value type | Description                                                                                                      |
+-----|-------------------------|------------|------------------------------------------------------------------------------------------------------------------|
+in   | cmd.status.get_report   | null       | Requests update progress information.                                                                            |
+out  | evt.status.report       | object     | Shows update progress and estimated time left. See [`ota_report_object`](#definitions) for detailed information. |
+in   | cmd.update.install      | string     | Starts an update of the device. Value should be the full path to the update file.                                | 
 
+## Definitions
+
+* `ota_report_object` is an object with the following structure:
+
+| Field          | Type   | Example         | Description                                                                                                     |
+|----------------|--------|-----------------|-----------------------------------------------------------------------------------------------------------------|
+| state          | string | `"IN PROGRESS"` | One of the `state` values.                                                                                      |
+| substate       | string | `"TRANSFERING"` | One of the `substate` values.                                                                                   |       
+| progress       | int    | `10`            | Optional, only available in substate `TRANSFERING`. Progress percentage of the update transfer.                 |
+| remaining_min  | int    | `3`             | Optional, only available in substate `TRANSFERING`. Remaining time until the end of the transfer (minutes).     | 
+| remaining_sec  | int    | `35`            | Optional, only available in substate `TRANSFERING`. Remaining time until the end of the transfer (seconds).     |
+| last_update    | int    | `132351314`     | Optional. Unix timestamp of the last started update process.                                                    |        
+
+
+* `state`: Current state of the update process, can be one of: `"NOT STARTED"`, `"IN PROGRESS"`, `"FINISHED"`.
+* `substate`: Gives more detailed information about the current update phase. Available values depend on the value in `state`:
+  * for `"IN PROGRESS"`: `"STARTED"`, `"TRANSFERING"`, `"REDISCOVERY"`, `"DEVICE REBOOTING"`, `"NEEDS USER ACTION TO START"`.
```

Comment:
I would resigne from predefined 'substate' enum to avoid adding more values in the future. Instead of this maybe we can have some string 'info' to print text in a generic way.
We could use it to report human-readable report errors, and substate etc ?

Reply:
This is what I suggested in the meeting. From my PoV it can all be squashed into following state:

NOT_STARTED
IN_RPOGRESS
NEEDS_USER_ACTION
FAILED (or FINISHED_WITH_ERRORS)
COMPLETED (or SUCCESSFUL or FINISHED_SUCCESSFULLY or COMPLETED_SUCCESSFULLY)

If we need more info we can add maybe a `reason` field for a reason why something failed, like this `LOW_BATTERY`, `INVALID_FILE`, etc.

Furthermore, all ENUMS should be non-whitespaced string. I see no reason for CAPITAL ENUMS as most other enums in FIMP documentation are simple snake case also.